lib/Doctrine/Common/Lexer/Token.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Common\Lexer;
+
+use ArrayAccess;
+use Countable;
+use Doctrine\Deprecations\Deprecation;
+use ReturnTypeWillChange;
+
+use function in_array;
+
+/**
+ * @template T of string|int
+ * @implements ArrayAccess<string,mixed>
+ */
+final class Token implements ArrayAccess, Countable
```

Comment:
:thinking: I will remove `Countable`, as `ArrayAccess` should be enough to easily implement a forward-compatibility layer.